python-cffi/netplan/netdef.py

```diff
@@ -242,20 +242,60 @@ def __next__(self):
         return ffi.string(next_value).decode('utf-8')
 
 
-@dataclass
+@dataclass(unsafe_hash=True)
 class NetplanRoute:
-    to: str
-    via: str
-    from_addr: str
-    type: str
-    scope: str
-    family: int
-    table: int
-    metric: int
-    mtubytes: int
-    congestion_window: int
-    advertised_receive_window: int
-    onlink: bool
+    _METRIC_UNSPEC_ = 2**32 - 1
+
+    to: str = None
+    via: str = None
+    from_addr: str = None
+    type: str = 'unicast'
+    scope: str = 'global'
+    protocol: str = None
+    table: str = 'main'
+    family: int = -1
+    metric: int = _METRIC_UNSPEC_
+    mtubytes: int = 0
+    congestion_window: int = 0
+    advertised_receive_window: int = 0
+    onlink: bool = False
+
+    def __str__(self):
+        route = ""
+        if self.to:
+            route = route + self.to
+        if self.via:
+            route = route + f' via {self.via}'
```

Comment:
thought: Hmm, what happens if `self.to` is empty. Our route string would start with some whitespace. Maybe we should `return route.strip()` at the end?